mantaray/received.py

```diff
@@ -484,6 +488,10 @@ def _handle_received_message(
     elif msg.command == RPL_TOPIC:
         _handle_numeric_rpl_topic(server_view, msg.args)
 
+    elif msg.command == RPL_WHOISUSER:
+        if msg.args[0] == server_view.core.nick:
+            server_view.core.nickmask = f"{msg.args[1]}!{msg.args[2]}@{msg.args[3]}"
```

Comment:
I would prefer to store only the `!user@host` part into the core, without the nickname. That way it's impossible to get a bug where the nick changes in either `core.nick` or `core.nickmask`, but not both. The rest of the mask will never change, unless you reconnect. We can then add a `get_user_mask()` method that returns the mask with `self.nick` in front.